cmd/svcat/broker/register_cmd.go

```diff
@@ -18,57 +18,122 @@ package broker
 
 import (
 	"fmt"
+	"os"
+	"strings"
+	"time"
 
 	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/command"
 	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/output"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	servicecatalog "github.com/kubernetes-incubator/service-catalog/pkg/svcat/service-catalog"
 	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // RegisterCmd contains the information needed to register a broker
 type RegisterCmd struct {
-	BrokerName string
-	Context    *command.Context
-	URL        string
+	*command.Waitable
+
+	Context *command.Context
+
+	BasicSecret       string
+	BearerSecret      string
+	BrokerName        string
+	CAFile            string
+	ClassRestrictions []string
+	PlanRestrictions  []string
+	SkipTLS           bool
+	RelistBehavior    string
+	RelistDuration    time.Duration
+	URL               string
 }
 
 // NewRegisterCmd builds a "svcat register" command
 func NewRegisterCmd(cxt *command.Context) *cobra.Command {
 	registerCmd := &RegisterCmd{
-		Context: cxt,
+		Context:  cxt,
+		Waitable: command.NewWaitable(),
 	}
 	cmd := &cobra.Command{
-		Use:   "register NAME --url URL",
+		Use:   "register NAME URL",
 		Short: "Registers a new broker with service catalog",
 		Example: command.NormalizeExamples(`
-		svcat register mysqlbroker --url http://mysqlbroker.com
+		svcat register mysqlbroker http://mysqlbroker.com
 		`),
 		PreRunE: command.PreRunE(registerCmd),
 		RunE:    command.RunE(registerCmd),
 	}
-	cmd.Flags().StringVar(&registerCmd.URL, "url", "",
-		"The broker URL (Required)")
-	cmd.MarkFlagRequired("url")
+	cmd.Flags().StringVar(&registerCmd.BasicSecret, "basic-secret", "",
+		"A secret containing basic auth (username/password) information to connect to the broker")
+	cmd.Flags().StringVar(&registerCmd.BearerSecret, "bearer-secret", "",
+		"A secret containing a bearer token to connect to the broker")
+	cmd.Flags().StringVar(&registerCmd.CAFile, "ca", "",
+		"A file containing the CA certificate to connect to the broker")
+	cmd.Flags().StringSliceVar(&registerCmd.ClassRestrictions, "class-restrictions", []string{},
+		"A whitelist of Classes to allow from the broker")
+	cmd.Flags().StringSliceVar(&registerCmd.PlanRestrictions, "plan-restrictions", []string{},
+		"A whitelist of Plans to allow from the broker")
+	cmd.Flags().StringVar(&registerCmd.RelistBehavior, "relist-behavior", "",
+		"Behavior for relisting the broker's catalogue. Manual/Duration")
```

Comment:
I think we use the US spelling "catalog" everywhere else, so let's rename "catalogue" to "catalog" for consistency.

Reply:
The pattern that we have used elsewhere for allowed values is:

* lower cased
* use a comma separated list of allowed values, examples below

https://github.com/kubernetes-incubator/service-catalog/blob/f31d3e56fde00dc6f9012675d10d7ccfafba1762/cmd/svcat/command/formatted.go#L49

https://github.com/kubernetes-incubator/service-catalog/blob/0a421721ee69ab5a8a89ecf279a326a163adcb4d/cmd/svcat/command/scoped.go#L59